console/console-init/ui/src/modules/device/dialogs/CreateDevice/AddGateway.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import {
+  Title,
+  TextInput,
+  FormGroup,
+  Form,
+  Button,
+  FlexItem,
+  Flex,
+  Chip,
+  ChipGroup,
+  GridItem,
+  Grid
+} from "@patternfly/react-core";
+import { DeviceListAlert } from "modules/device/components";
+import { deviceIDRegExp } from "utils";
+
+export interface IGatewayProps {
+  addedGateways: string[];
+  removeGateway: (id: string) => void;
+  appendGateway: (deviceId: string) => void;
+}
+
+export const AddGateway: React.FunctionComponent<IGatewayProps> = ({
+  addedGateways,
+  removeGateway,
+  appendGateway
+}) => {
+  const [inputID, setInputID] = useState<string>();
+  const [isValid, setIsValid] = useState<boolean>(false);
+
+  const onHanldeClick = () => {
```

Comment:
Event name is not clear. can you please revisit. It should be something  onClick+handlername 
suggestion like onClickGatewayButton.